extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationOnSiteNotifications.js

```diff
@@ -1,42 +1,141 @@
 require(
-	['jquery', 'wikia.window', 'wikia.mustache'],
-	function ($, window, mustache) {
+	['jquery', 'wikia.window', 'wikia.loader', 'wikia.mustache'],
+	function ($, window, loader, mustache) {
 		'use strict';
 
 		var OnSiteNotifications = {
-			init: function () {
+			init: function (template) {
+				this.template = template;
 				this.bucky = window.Bucky('OnSiteNotifications');
 
 				// we only want 1 update simultaneously
 				this.updateInProgress = false;
 
-				setTimeout(this.proxy(this.updateCounts), 300);
+				setTimeout(this.proxy(this.updateUnreadCount), 300);
 
 				this.$window = $(window);
-				this.$notificationsCount = $('.on-site-notifications-count');
+				this.$notificationsCount = $('#on-site-notifications-count');
+				this.$container = $('#on-site-notifications');
+
+				this.addDropdownLoadingEvent();
+				this.addMarkAllAsReadEvent();
+			},
+
+			addMarkAllAsReadEvent: function () {
+				this.$markAllAsReadButton = $('#mark-all-as-read-button');
+				this.$markAllAsReadButton.click(function () {
+					OnSiteNotifications.markAllAsRead();
+				});
+			},
+
+			addDropdownLoadingEvent: function () {
+				var $dropdown = $('#on-site-notifications-dropdown');
+				$dropdown.click(function () {
+					OnSiteNotifications.loadFirstPage();
+				});
 			},
 
-			updateCounts: function () {
+			shouldLoadFirstPage: function () {
+				return !this.updateInProgress && this.nextPage && this.allPagesLoaded !== true;
+			},
+
+			shouldLoadNextPage: function () {
```

Comment:
Oh it's going to be, trust me I'm doing this the second time, just differently. =)